packages/create/template.js

```diff
@@ -0,0 +1,168 @@
+"use strict";
+const { logger } = require("./logging");
+const fs = require("fs-extra");
+const path = require("path");
+
+async function copyTemplateFile(src, dst, properties) {
+	logger.verbose(`Writing ${dst} from template ${src}`);
+
+	// Closure, get the property value or throw
+	function getProperty(property) {
+		if (!properties.hasOwnProperty(property)) {
+			throw new Error(`Unknown property (${property}) found in template file ${src}`);
+		}
+
+		return properties[property];
+	}
+
+	// Closure, Get the boolean conjunction of a set of properties
+	function propertyConjunction(props) {
+		for (const property of props) {
+			if (property.startsWith("!")) {
+				if (Boolean(getProperty(property.slice(1)))) {
+					return false;
+				}
+			} else if (!Boolean(getProperty(property))) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	// Process the the template file
+	const badChars = /[\(\)-+/*<>=]/;
+	const templateData = await fs.readFile(src, "utf8");
+	const outputData = templateData
+		// match preprocessor comments: //%//
+		.replace(/^\/\/%\/\/.*\n/gm, "")
+		// match preprocessor insert: __property__
+		.replace(/__([a-zA-Z][a-zA-Z0-9_]*?[a-zA-Z0-9]?)__/gm, (_, property) => getProperty(property))
+		// match preprocessor conditionals: //%if <CONDITION> [// COMMENT]\n <CONTENT> \n//%endif [COMMENT]\n
+		.replace(/^\/\/%if ([^\n\/]*)(?:\/\/.*?)?\n((?:.|\n)*?)\/\/%endif.*\n/gm, (_, condition, content) => {
+			let include = false;
+			const conjunctions = condition.split("|");
+			for (const conjunct of conjunctions) {
+				if (badChars.test(conjunct)) {
+					throw new Error("Preprocessor condition can only contain logical operators" +
+						"such as & | or ! it can not have arithmetic, comparisons or parenthesis"
+					);
+				}
+
+				const property_names = conjunct.split("&").map(s => s.trim());
+				if (propertyConjunction(property_names)) {
+					include = true;
+					break;
+				}
+			}
+
+			return include ? content : "";
+		});
+
+	// Attempt to write the output file, warn if it already exists
+	try {
+		await fs.outputFile(dst, outputData);
+	} catch (err) {
```

Comment:
This will truncate the file (wipe it's content) and write the outputData to it if it already exists because the default flag is `"w"` (see the meaning of the flags [here](https://nodejs.org/api/fs.html#file-system-flags)).
```suggestion
	try {
		await fs.outputFile(dst, outputData, { flag: "wx" });
	} catch (err) {
```